pallets/capacity/src/lib.rs

```diff
@@ -528,7 +528,11 @@ impl<T: Config> Pallet<T> {
 		target: MessageSourceId,
 		target_details: StakingTargetDetails<BalanceOf<T>>,
 	) {
-		StakingTargetLedger::<T>::insert(staker, target, target_details);
+		if target_details.amount.is_zero() {
```

Comment:
note that due to rounding errors, target_details.amount may be 1.  There's no need to check it, regardless.